### ISSUE TITLE
Jython compatibility

### ISSUE THREAD

**Issue by frenzymadness** [python-attrs/attrs#321]
Hello.

I've tried to use attrs module with Jython but it seems to be unsupported. I have an empty docker container based on Fedora 27 and Jython 2.7.1 (latest stable) installed from Fedora repository and attrs installed from PyPI.

Steps to reproduce:
```
$ jython -m ensurepip
Collecting setuptools
Collecting pip
Installing collected packages: setuptools, pip
Successfully installed pip-9.0.1 setuptools-28.8.0
$ jython -m pip install attrs
Collecting attrs
  Downloading attrs-17.4.0-py2.py3-none-any.whl
Installing collected packages: attrs
Successfully installed attrs-17.4.0
$ jython 
Jython 2.7.1 (, Sep 29 2017, 12:00:52) 
[OpenJDK 64-Bit Server VM (Oracle Corporation)] on java1.8.0_151
Type "help", "copyright", "credits" or "license" for more information.
>>> import attr
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/share/jython/Lib/site-packages/attr/__init__.py", line 5, in <module>
    from . import converters, exceptions, filters, validators
  File "/usr/share/jython/Lib/site-packages/attr/filters.py", line 7, in <module>
    from ._compat import isclass
  File "/usr/share/jython/Lib/site-packages/attr/_compat.py", line 139, in <module>
    set_closure_cell = make_set_closure_cell()
  File "/usr/share/jython/Lib/site-packages/attr/_compat.py", line 131, in make_set_closure_cell
    set_closure_cell = ctypes.pythonapi.PyCell_Set
AttributeError: 'module' object has no attribute 'pythonapi'
```

Is there any chance, that attrs would support Jython? I am asking because pytest is considered supported on Jython but depends on attrs so pytest also doesn't work.

Thank you and have a nice day!

**Comment by The-Compiler** [comment]
Looks like the related code was added in #226 for #102 (cc @Tinche)

**Comment by Tinche** [comment]
We could add Jython as a special case. Does Jython support changing closure cell contents a different way?

**Comment by frenzymadness** [comment]
@Tinche Sorry, I don't know Jython enough to answer your question.

**Comment by RonnyPfannschmidt** [comment]
@jeff5 should know more details

**Comment by hynek** [comment]
Jython is Python 2-only, so we can just not set a closure cell and be done with it.

**Comment by jeff5** [comment]
@RonnyPfannschmidt : not really. As far as I can tell, any use of ``ctypes`` in our tests is guarded by a skip-if-jython. It exists as a module, but maybe just to satisfy imports or some very limited uses.

I haven't fully understood what you're doing with ``ctypes.pythonapi``. Obviously you can't expect the low-level C-API of CPython to work in a Java implementation. ``cell`` is an exposed type, of course, and my naive attempts to write its ``ob_ref`` member (which is public) are thwarted by coercion. I got most of the way with:
```
>>> c = g.func_closure[0]
>>> ref = java.lang.Class.getField(org.python.core.PyCell, "ob_ref")
>>> ref.get(c)
5
```
where ``g`` is a function defined in an inner scope, but in ``ref.set(c, 6)`` the 6 is coerced to a ``java.lang.Integer`` so cannot be assigned.

Any ideas @jimbaker ?

**Comment by hynek** [comment]
What we’re doing is enabling things like bare supers and __class__ to work.   AFAICT there’s no downside to skip this on Python 2, so we could add a case for Jython.

Sadly, I have no way to test it because pyenv just exploded when trying to install it.

Someone just has to add a case to https://github.com/python-attrs/attrs/blob/1f0d85199666cfa881385c31314a6b2d39f6c36e/src/attr/_compat.py#L121 making set_closure_cell a NOP.

**Comment by hynek** [comment]
This is fixed in master and goes into attrs 18.1.
